src/domain/safe/entities/transfer.entity.ts

```diff
@@ -12,11 +11,43 @@ export interface ERC20Transfer extends Transfer {
   value: string;
 }
 
+export function isERC20Transfer(transfer: Transfer): transfer is ERC20Transfer {
+  const erc20Transfer = transfer as ERC20Transfer;
+  return (
+    erc20Transfer.tokenAddress !== undefined &&
+    erc20Transfer.tokenAddress !== null &&
+    erc20Transfer.value !== undefined &&
+    erc20Transfer.value !== null
+  );
+}
+
 export interface ERC721Transfer extends Transfer {
   tokenAddress: string | null;
   tokenId: string;
 }
 
+export function isERC721Transfer(
+  transfer: Transfer,
+): transfer is ERC721Transfer {
+  const erc721Transfer = transfer as ERC721Transfer;
+  return (
+    erc721Transfer.tokenAddress !== undefined &&
+    erc721Transfer.tokenAddress !== null &&
```

Comment:
Similar to the above ☝️ 

I think our model is incorrect. `tokenAddress` cannot be `null`.

After changing that you can write it like:

```typescript
erc721Transfer.tokenAddress != null
```

Reply:
[a41f860](https://github.com/5afe/safe-client-gateway-nest/pull/206/commits/a41f8600dc9fbe237de992a74c51dce6a6703668)